mermaid_py/client.py

```diff
@@ -0,0 +1,443 @@
+        information.
+        :param name: (optional if 'id' provided) MERMAID project name.
+        :type name: str
+        :param id: (optional if 'name' provided) MERMAID project ID.
+        :type: id: str
+        :param resource: MERMAID project entity resources. /projects/<project_id>/<resource>
+        :param filter: MERMAID project resource filters.
+        :type filter: str
+        :param filter_val: (optional) Required for resource filters requiring values. eg. (size_min/size_max,
+        count_min/count_max, length_min/length_max, len_surveyed_min/len_surveyed_max,
+        sample_date_before/sample_date_after).
+        :type filter_val: str, int
+        :return: project resource.
+        :rtype: dict
+        """
+        # Generate path.
+        path = None
+        if resource:
+            if id:
+                path = f"projects/{id}/{resource}/"
+            elif name:
+                p_id = self.get_project_id(name=name)
+                # Nested if instead of "elif name and self.get_project_id(name=name)"
+                # to improve performance by reducing the number of API calls for only 'id' NoneType check.
+                if p_id:
+                    path = f"projects/{p_id}/{resource}/"
+
+        # No resource given generate path
+        else:
+            if id:
+                path = f"projects/{id}/"
+            # Get id for name of project and create path
+            elif name:
+                p_id = self.get_project_id(name=name)
+                if p_id:
+                    path = f"projects/{p_id}/"
+
+        # API call from generated path and valid filters check.
+        if path:
+            if filter and filter_val:
+                payload = {filter: filter_val}
+                return self._fetch_resource(path, parameters=payload)
+            elif filter:
+                return self._fetch_resource(path, parameters=filter)
+            else:
+                return self._fetch_resource(path)
+        return self._fetch_resource(path)
+
+    # Get functions.
+    def get_info(self, info):
+        """
+        Gets non-project resource information. Authentication not required for health, version, profiles, projecttags,
+        sites, managements. Authentication required for me.
+        See https://mermaid-api.readthedocs.io/en/latest/nonproject.html for more information on non-project resources.
+        :param info: health, version, profiles, projecttags, sites, managements, me.
+        :type info: str
+        :return: non-project resource data.
+        :rtype: dict, str
+        """
+        # TODO: Handle pagination
+        if info == "me" and not self.authenticated:
+            raise Exception('Authentication Required for access to "me" endpoint')
+
+        if info in self.npr_endpoints:
+            return self._fetch_resource(info)
+        else:
+            raise Exception(f"get_info() invalid info parameter:{info}")
```

Comment:
We shouldn't raise generic `Exception`s -- we should create our own, in a separate exceptions file. I would call this something like `InvalidResourceException`. We can handle this in a separate card if you'd rather (in which case create one before resolving this).